stl/inc/memory

```diff
@@ -3079,15 +3079,21 @@ inline void* align(size_t _Bound, size_t _Size, void*& _Ptr, size_t& _Space) noe
     return _Ptr;
 }
 
-#if _HAS_CXX20 && 0
+#if _HAS_CXX20
 template <size_t _Nx, class _Ty>
 _NODISCARD constexpr _Ty* assume_aligned(_Ty* _Ptr) noexcept /* strengthened */ {
-    // this enforces the requirement that _Nx be a power of two
-    __builtin_assume_aligned(_Ptr, _Nx);
+#if 1 // TRANSITION: delete after __builtin_assume_aligned will be constexpr
+    if (!std::is_constant_evaluated()) {
+        return static_cast<_Ty*>(__builtin_assume_aligned(_Ptr, _Nx));
+    }
 
     return _Ptr;
+#else
+    // this enforces the requirement that _Nx be a power of two
+    return static_cast<_Ty*>(__builtin_assume_aligned(_Ptr, _Nx));
+#endif // TRANSITION: delete after __builtin_assume_aligned will be constexpr

```

Comment:
1. We conventionally use the `_STD` macro in product code, which expands to `::std::`. This theoretically defends against pathological user namespaces. (Even though `is_constant_evaluated()` takes no arguments and is therefore invulnerable to argument-dependent lookup hijacking.)
2. We conventionally prefer to avoid negated tests, unless attempting to do so over-complicates the code. This is because it is generally easier to read and understand positive tests ("if cat", instead of "if not cat"). In this case, I believe that flipping this around to `if (_STD is_constant_evaluated())` guarding `return _Ptr;` would have a bonus simplifying effect - that branch could be marked as `TRANSITION`, and then the following usage of `__builtin_assume_aligned` wouldn't need to be repeated (in active and inactive forms).

Reply:
I read your explanation about `_STD` here and in discord. And I made the same mistake :(
Sorry you had to repeat this for the thousandth time.

Reply:
Heh, no problem - it takes a long time to get used to our unusual conventions!